addOns/paramminer/src/test/java/org/zaproxy/addon/paramminer/UrlUtilsUnitTest.java

```diff
@@ -0,0 +1,142 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.testutils.NanoServerHandler;
+import org.zaproxy.zap.testutils.TestUtils;
+
+class UrlUtilsUnitTest extends TestUtils {
+
+    @BeforeEach
+    void init() throws Exception {
+        setUpZap();
+        try (InputStream is =
+                UrlUtilsUnitTest.class.getResourceAsStream("/wordlists/small_list.txt")) {
+            Path file = Paths.get(Constant.getZapHome() + "/wordlists/small_list.txt");
+            Files.createDirectories(file.getParent());
+            Files.copy(is, Paths.get(Constant.getZapHome() + "/wordlists/small_list.txt"));
+        }
+        startServer();
+    }
+
+    @AfterEach
+    void teardown() {
+        stopServer();
+    }
+
+    @Test
+    void shouldLoadFileWithoutErrors() throws IOException {
+        String path = Paths.get(Constant.getZapHome() + "/wordlists/small_list.txt").toString();
+        List<String> params = UrlUtils.read(path);
+        assertEquals(params.size(), 835);
+        assertEquals(params.get(10), "Id");
+    }
+
+    @Test
+    void shouldGetStableMessages() {
+        String path = "/example/";
+        String url = "http://localhost:" + this.nano.getListeningPort() + path;
+        HttpRequestHeader header = new HttpRequestHeader();
+        header.addHeader("X-forwarded-host", "local.com");
+
+        this.nano.addHandler(
+                new NanoServerHandler(path) {
+                    @Override
+                    protected Response serve(IHTTPSession session) {
+                        String q = getFirstParamValue(session, "q");
+                        return newFixedLengthResponse(
+                                "<html><body>Hello " + q + " World!</body></html>");
+                    }
+                });
+        HttpMessage msg = UrlUtils.stableRequest(url, header);
+        assertEquals(msg.getResponseHeader().getStatusCode(), 200);
+        assertEquals(msg.getRequestHeader().getHeader("X-forwarded-host"), "local.com");
+        assertEquals(msg.getResponseHeader().getHeader("X-forwarded-host"), null);
+    }
+
+    @Test
+    void shouldPopulateMapWithoutErrors() {
+        String path = Paths.get(Constant.getZapHome() + "/wordlists/small_list.txt").toString();
+        List<String> params = UrlUtils.read(path);
+        Map<String, String> map = UrlUtils.populate(params);
+        assertEquals(map.size(), 835);
+        assertEquals(map.get("Id"), "111110");
+        assertEquals(map.get("filter_relational_operator_10"), "111288");
+    }
+
+    @Test
+    void shouldSliceWithoutErrors() {
+        String path = Paths.get(Constant.getZapHome() + "/wordlists/small_list.txt").toString();
```

Comment:
Better to use a test file, otherwise the tests will probably fail if the list is refined.

Reply:
a mini list you say?

Reply:
put it in zapHomeFiles?

Reply:
You could allow setting the list just for test purposes. Ummmm have a look at the suspicious comments passive scan rule (I think that's a good example....if I recall correctly).

Reply:
Should go in `src/test/resources`, the code will be the same it just changes the resource path when copying.